sub-packages/bionemo-scdl/tests/bionemo/scdl/io/test_single_cell_memmap_dataset.py

```
@@ -312,6 +312,18 @@ def test_lazy_load_SingleCellMemMapDatasets_another_dataset(tmp_path, compare_fn
     compare_fn(ds_regular, ds_lazy)
 
 
+def test_lazy_load_SingleCellMemMapDatasets_with_raw(tmp_path, compare_fn, make_h5ad_with_raw):
+    h5ad_path = make_h5ad_with_raw(tmp_path)
+    ds_regular = SingleCellMemMapDataset(tmp_path / "regular_load", h5ad_path=h5ad_path)
+    ds_lazy = SingleCellMemMapDataset(
+        tmp_path / "lazy_load",
+        h5ad_path=h5ad_path,
+        paginated_load_cutoff=0,
+        load_block_row_size=2,
+    )
+    compare_fn(ds_regular, ds_lazy)
+
```

Could we also add (pseudocode):

```
ad = anndata.load(h5ad_path)
assert False == compare_fn(h5ad_path.raw, h5ad_path.X) 
```

as a reader, I want to see that this doesnt silently fallback on .X by reading the test, but to do so, I need to look through comapre_fn and make_h5ad_with_raw to see whats actually happening. The above assertion makes it abundantly clear the conditions being checked.

---

will do!